tests/AppSwitcher.test.js

```diff
@@ -0,0 +1,234 @@
+import userEvent from "@testing-library/user-event";
+import { getByText, render } from "@testing-library/react";
+
+import { AppSwitcher, Sidebar } from "../lib/components/layouts";
+import { STORYBOOK_NAV_LINKS } from "../example/src/constants";
+
+const AppSwitcherTest = ({ children }) => {
+  const [isSidebarCollapsed, setIsSidebarCollapsed] = useState(true);
+
+  return (
+    <Router>
+      <Sidebar
+        navLinks={STORYBOOK_NAV_LINKS.slice(3)}
+        onAppSwitcherToggle={() => {}}
+        isCollapsed={isSidebarCollapsed}
+        onCollapse={() => setIsSidebarCollapsed((collapsed) => !collapsed)}
+        showAppSwitcher
+        appName="neetoUI"
+        profileInfo={{
+          name: "John Doe",
+          email: "john@doe.com",
+          topLinks: [
+            {
+              label: "Logout",
+            },
+            {
+              label: "Settings",
+            },
+          ],
+        }}
+      />
+      {children}
+    </Router>
+  );
+};
+
+const neetoApps = ["KB", "Desk", "Planner"];
+
+describe("AppSwitcher", () => {
+  it("should render without error", () => {
+    const { getByText } = render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="KB"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    for (const app of neetoApps) {
+      expect(getByText(app)).toBeInTheDocument();
+    }
+  });
+
+  it("should not render when isOpen is false", () => {
+    const { queryByText } = render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen={false}
+          neetoApps={neetoApps}
+          activeApp="KB"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    for (const app of neetoApps) {
+      expect(queryByText(app)).not.toBeInTheDocument();
+    }
+  });
+
+  it("should render active app correctly", () => {
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+    const neetoUIClassPrefix = "neeto-ui-app-switcher-link--";
+    const activeLink = document.querySelector(`.${neetoUIClassPrefix}active`);
+    expect(activeLink).toBeInTheDocument();
+    expect(getByText(activeLink, "Planner")).toBeInTheDocument();
+  });
+
+  it("should render different sizes correctly", () => {
+    const size = "lg";
+    render(
+      <AppSwitcherTest>
+        <AppSwitcher
+          isOpen
+          neetoApps={neetoApps}
+          activeApp="Planner"
+          size={size}
+          environment={process.env.NODE_ENV}
+        />
+      </AppSwitcherTest>
+    );
+
+    const neetoUIClassPrefix = "neeto-ui-app-switcher__wrapper-size--";
+    expect(
+      document.querySelector(`.${neetoUIClassPrefix}${size}`)
+    ).toBeInTheDocument();
+  });
```

Comment:
I think this would be taken care of by the chromatic visual testing. You can remove this test.

Reply:
We would only want to test the interactivity and the rendering part in the unit tests. Do not test for styles.